### DISCUSSION TITLE
Ki editor

### DISCUSSION THREAD

**Discussion by ilyagr** [71/dance#347] [General]
Just saw a new editor with an interesting modal system that seems even more organized than Kakoune's.

https://news.ycombinator.com/item?id=41484503

To summarize, the idea is that every command has the form ["selection mode" -> "movement" -> "action"](https://ki-editor.github.io/ki-editor/comparisons/modal-editors.html) instead of Kakoune's movement->action.

So, instead of having separate commands for "next character", "next word", "next structural element", there is one command for "next" (as well as [a few additional motions](https://ki-editor.github.io/ki-editor/normal-mode/core-movements.html)). Every command starts with a ["selection mode"](https://ki-editor.github.io/ki-editor/normal-mode/selection-modes/index.html) like "column", "line", or "structural element".

**Comment by Strackeror** [comment]
I saw the post on hacker news, it's a very interesting take on modal editing ! 
At a glance it looks very possible to implement the basic idea within dance.  Support for https://github.com/71/vscode-tree-sitter-api would need to be improved for the syntax tree related modes, and I suspect there's a few quirks where vscode's extension API would be limiting, but it's definitely something I'd be interested seeing some experimentation in.
